js/main.js

```diff
@@ -5,33 +5,82 @@ var $formSubmit = document.querySelector('form');
-
-// Listening for input changes for IMAGE
+var $ul = document.querySelector('ul');
 $imgUrl.addEventListener('input', changeImg);
 function changeImg(event) {
   $setImg.setAttribute('src', event.target.value);
 }
 
-// Listening for submit event
 $formSubmit.addEventListener('submit', logForm);
 function logForm(event) {
   event.preventDefault();
   var formObject = {
     title: $inputTitle.value,
     photoUrl: $imgUrl.value,
     notes: $textArea.value,
-    nextEntryId: data.nextEntryId + 1
+    nextEntryId: data.nextEntryId + 1,
+    view: 'entry-form',
   };
   data.nextEntryId += 1;
   data.entries.unshift(formObject);
   $setImg.setAttribute('src', 'images/placeholder-image-square.jpg');
   $formSubmit.reset();
+  //Updating DOM as SAVE is clicked
+  $ul.prepend(entryDOM(formObject));
+  $newEntry.className = 'hidden';
+  $entriesPage.className = '';
+  data.view = "all-entries"
+}
+
+// Issue 2 - User can view their entries
+
+function entryDOM(entry) {
+  var $createLi = document.createElement('li');
+  var $createDivData = document.createElement('div');
+  var $createDivRow = document.createElement('div');
+  var $createImg = document.createElement('img');
+  var $createDivColHalf = document.createElement('div');
+  var $createH2 = document.createElement('h2');
+  var $createP = document.createElement('p');
+
+  $createLi.appendChild($createDivData);
+  $createDivData.setAttribute('data-view', 'entries');
+  $createDivData.appendChild($createDivRow);
+  $createDivRow.setAttribute('class', 'row');
+  $createDivRow.appendChild($createImg);
+  $createImg.setAttribute('src', entry.photoUrl);
+  $createImg.setAttribute('height', '400px');
+  $createImg.setAttribute('class', 'column-half');
+  $createImg.setAttribute('class', 'column-half');
+  $createDivRow.appendChild($createDivColHalf);
+  $createDivColHalf.setAttribute('class', 'column-half');
+  $createDivColHalf.appendChild($createH2);
+  $createH2.textContent = entry.title;
+  $createDivColHalf.appendChild($createP);
+  $createP.textContent = entry.notes;
+  $ul.appendChild($createLi);
+
+  return $createLi
 }
 
-// Put the form 's input values into a new object.
-// Add the nextEntryId to the object.
-// Increment the nextEntryId on the data model.
-// Prepend the new object to the entries in the data model.
-// Reset the image preview 's `src'
-// attribute.
-// Reset the form inputs.
+var $entriesPage = document.querySelector('div[data-view="entries"]');
+var $newButton = document.querySelector('.newButton');
+var $newEntry = document.querySelector('div[data-view="entry-form"]');
+var $entriesLink = document.querySelector('a[href="#entry-form"]');
+
+$newButton.addEventListener('click', function () {
+  $newEntry.className = '';
+  $entriesPage.className = 'hidden';
+  data.view = "entry-form"
+});
+
+$entriesLink.addEventListener('click', function () {
+  $newEntry.className = 'hidden';
+  $entriesPage.className = '';
+  data.view = "all-entries"
```

Comment:
Instead of doing each of these manually, I would make one function definition that receives one argument of a string containing the name of the view to switch to, then loop over all views and show the view that matches that name, and hide all the other views that don't match. That way you can easily add new views in the future without needing to write new code.

Reply:
I would name the function `switchViews`